src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml

```diff
@@ -5,15 +5,20 @@
              Background="White"
              mc:Ignorable="d" d:DesignWidth="400" d:DesignHeight="450"
              x:Class="DotnetCountersUi.Views.GraphToolbox"
-             x:Name="Ctrl">
-    <Grid ColumnDefinitions="*,Auto,*">
-        <TextBlock Grid.Column="1" Text="(unnamed)" VerticalAlignment="Center" />
-        <StackPanel Grid.Column="2" Orientation="Horizontal" Spacing="4" Margin="4, 0">
-            <Button Height="25" Width="25" HorizontalAlignment="Right" VerticalAlignment="Top"
-                    Command="{Binding #Ctrl.DeleteCommand}"
-                    CommandParameter="{Binding #Ctrl.DeleteCommandParameter}">
-                <PathIcon Data="{StaticResource delete_regular}" />
-            </Button>
-        </StackPanel>
+             x:Name="Ctrl"
+             Padding="30, 0">
+    <Grid ColumnDefinitions="25,*,Auto,*,25">
+        <TextBox Name="RenameBox" Grid.Column="2" Text="{Binding #Ctrl.Text}" IsVisible="False"
+                 KeyUp="RenameBox_OnKeyUp" LostFocus="RenameBox_OnLostFocus" />
+        <TextBlock Name="NameBlock" Grid.Column="2" Text="{Binding #Ctrl.Text}" VerticalAlignment="Center" />
```

Comment:
For some reason, `TextBox` no longer shrinks correctly. If you rename your graph to `"A"`, the editable `TextBox` will be quite large, but when the editing is finished, the `TextBlock` shrinks to one-letter size. Maybe there can be a minimum width for both controls to synch their behavior?

Reply:
Looks better now, thanks!